pytorch_lightning/trainer/training_loop.py

```diff
@@ -181,13 +181,12 @@ def on_train_end(self):
         if self.trainer.global_rank == 0:
             self.trainer.profiler.describe()
 
-        if self.trainer.global_rank == 0:
-            for proc in self.trainer.interactive_ddp_procs:
-                subprocess.Popen.kill(proc)
+        # give accelerators a chance to finish
+        self.trainer.accelerator_backend.on_train_end()
 
         # clean up dist group
-        if self.trainer.use_ddp or self.trainer.use_ddp2:
-            torch_distrib.destroy_process_group()
+        # if self.trainer.use_ddp or self.trainer.use_ddp2:
```

Comment:
where is this done now?

Reply:
it’s not. it was causing all sorts of issues and not needed. 

ie: .fit then .test didn’t work with this because .fit turned everything off

Reply:
but then the processes will never be terminated, right?
so what happens if user does not call `.test` afterwards? The processes won't be terminated and live on forever. I think, we could probably add this to trainers/backends `__del__`

Reply:
it terminates when the process terminates (the script).
and ddp can only ever be called via a script anyhow not a notebook or something